packages/factorywoman/client/factoryclosure.js

```diff
@@ -0,0 +1,53 @@
+FactoryClosure = {};
+
+'use strict';
+
+FactoryClosure = function(func, callback, funcCount) {
+  var self = this;
+
+  this._callback = callback;
+  this._funcCount = funcCount;
+  this._counter = 0;
+  this._traitCounter = 0;
+  this._stack = [];
+  this._interval = setInterval(function() {
+    self.closureWorker();
+  }, 50);
```

Comment:
Because nested traits do not get the chance to get executed properly. Emphasizes that this is unfortunately still highly dependent on the current performance of your PC.